### PR TITLE
[Dictionary] Fix displaying of dictionary with > 12 elements on 32-bits.

### PR DESCRIPTION
**Pull request by dcci**
<rdar://problem/47271307>

### CODE REVIEW

**Comment by dcci** [comment]
This fixes TestSwiftStdlibDictionary.py on 32-bits.

**Comment by dcci** [comment]
@swift-ci test and merge

File: source/Plugins/Language/Swift/SwiftHashedContainer.cpp [hunk 1]
```diff
@@ -493,8 +493,6 @@ NativeHashedStorageHandler::NativeHashedStorageHandler(
   if (!scale_sp)
     return;
   auto scale = scale_sp->GetValueAsUnsigned(0);
-  if (scale > m_ptr_size)
-    return;
```
Comment (lorentey) [inline source/Plugins/Language/Swift/SwiftHashedContainer.cpp:497]: Hm, I don't think we found out why this was failing. Scales larger than 31/63 would wreak havoc in `GetBucketCount`:

```c++
  // Line 88
  size_t GetBucketCount() { return 1 << m_scale; }
```

(Related: this should probably be changed to read `1ULL << m_scale`, as well.)

Comment (dcci) [reply source/Plugins/Language/Swift/SwiftHashedContainer.cpp:497]: This is `m_ptr_size` in /bytes/, i.e. 4. Should we have a check for this (but in bits?)

Comment (dcci) [reply source/Plugins/Language/Swift/SwiftHashedContainer.cpp:497]: i.e. `(scale > m_ptr_size * 8)`

Comment (lorentey) [reply source/Plugins/Language/Swift/SwiftHashedContainer.cpp:497]: Argh, of course. 🤭 That's simultaneously a relief and a huge face palm.

This wasn't a super important condition -- it was just a sanity check against trying to interpret corrupt data. (In a valid Set/Dictionary, the scale should be between 0 and 32/64, depending on pointer size in bits.) I think we can safely omit it; subsequent code will fail when it tries to access memory anyway.

Comment (dcci) [reply source/Plugins/Language/Swift/SwiftHashedContainer.cpp:497]: Great :)

**Pull request merged**